k2/csrc/array.h

```diff
@@ -237,8 +237,32 @@ class Array1 {
     return ans;
   }
 
-  // Copy from another array of the same dimension and type.
-  void CopyFrom(const Array1<T> &src);
+#define ToType(type, name)                                    \
```

Comment:
I am using macros instead of templates here because it throws the following error
```
error: ‘__T0’ was not declared in this scope
```
if templates were used here. I don't know how to fix such kinds of errors, so macros are used here.

Reply:
M, OK, please make a comment so it's quite clear what is going on.

Reply:
ok, I just added some documentation to the macro `ToType` explaining what it is doing.